logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -1,38 +1,35 @@
 package com.wire.kalium.logic.data.message
 
 import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationId
 import com.wire.kalium.logic.data.id.IdMapper
-import com.wire.kalium.logic.data.id.QualifiedID
 import com.wire.kalium.logic.functional.Either
 import com.wire.kalium.network.api.message.MessageApi
-import com.wire.kalium.persistence.dao.message.Message
 import com.wire.kalium.persistence.dao.message.MessageDAO
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 
 interface MessageRepository {
-    suspend fun getMessagesForConversation(conversationId: QualifiedID, limit: Int): Flow<List<Message>>
+    suspend fun getMessagesForConversation(conversationId: ConversationId, limit: Int): Flow<List<Message>>
     suspend fun persistMessage(message: Message): Either<CoreFailure, Unit>
 }
 
 class MessageDataSource(
-    private val idMapper: IdMapper,
     private val messageApi: MessageApi,
-    private val messageDAO: MessageDAO
+    private val messageDAO: MessageDAO,
+    private val messageMapper: MessageMapper,
+    private val idMapper: IdMapper
 ) : MessageRepository {
 
-    override suspend fun getMessagesForConversation(conversationId: QualifiedID, limit: Int): Flow<List<Message>> {
-        return messageDAO.getMessageByConversation(idMapper.toDaoModel(conversationId), limit)
+    override suspend fun getMessagesForConversation(conversationId: ConversationId, limit: Int): Flow<List<Message>> {
+        return messageDAO.getMessageByConversation(idMapper.toDaoModel(conversationId), limit).map { messageList ->
+            messageList.map(messageMapper::fromRecordToMessage)
```

Comment:
Which reflection issues you mean?

You mean the code being deleted by minification? This would not be the case. There is code referencing the function.